### PR TITLE
Lint cleanup

### PR DESCRIPTION
**Pull request by sreynen**
This resolves all `npm run lint` warnings and errors, so we can eventually use lint as a basic check on new code. The commits are broken up into error type, for hopefully easier review. Most errors were resolved with simple changes that almost certainly won't impact code functionality. But a few errors involved more complex code changes that may have inadvertently impacted functionality, so definitely worth reviewing this closely. And some errors were unavoidable due to external dependencies, so those were resolved by adding `eslint-disable-next-line` comments.

### CODE REVIEW

**Review by shakalee14 — COMMENTED** [review]
this does not break on my local environment - LGTM!

File: src/components/AdminDashboard.jsx [hunk 1]
```diff
@@ -73,7 +71,7 @@ class AdminDashboard extends React.Component {
       <div>
         <TopNav title={title} backToURL={backToURL} />
         <div className={css(styles.container)}>
-          {this.renderNavigation(sections)}
+          {this.renderNavigation()}
```
[the remark this answers is not in the document]
Comment (sreynen) [reply src/components/AdminDashboard.jsx:74]: We were previously passing `sections` to `renderNavigation()`, but never using it within that function. ESLint flagged the unused variable, so to resolve that, I removed the parameter from both the function definition and the call.

**Comment by shakalee14** [comment]
this is mergeable @sreynen

**Comment by shakalee14** [comment]
@sreynen did you want to resolve these conflicts?

**Comment by sreynen** [comment]
When I have the time, yes.

**Pull request merged**